app/src/main/java/eu/kanade/tachiyomi/data/backup/create/creators/MangaBackupCreator.kt

```diff
@@ -98,4 +98,6 @@ private fun Manga.toBackupManga() =
         updateStrategy = this.updateStrategy,
         lastModifiedAt = this.lastModifiedAt,
         favoriteModifiedAt = this.favoriteModifiedAt,
+        version = this.version,
+        isSyncing = this.isSyncing,
```

Comment:
I've been out of the loop on the syncing logic for a while. What is `isSyncing` used for?

Reply:
It's to ensure consistency across devices during sync operations. This flag plays a crucial role in preventing automatic version increments by our triggers during the sync process. Without this flag, the act of syncing itself would inadvertently increment the manga's version number, leading to inconsistencies across devices.

Here's a practical scenario to illustrate why the is_syncing flag is essential: Imagine two devices, A and B, where Device A is on version 6 and Device B on version 7 of a manga. Ideally, syncing Device A should update its manga version to match Device B's version 7. However, without the is_syncing flag acting as a safeguard, the sync process would unintentionally increment the version to 8. This increment during syncing could lead to a situation where, if Device B holds the latest version and subsequent changes are made on Device A, those changes might not be recognized appropriately due to the version discrepancy.

Reply:
In that case it's not necessary to make it a part of the Manga/Chapter model is it? You can just make it a parameter of the insert function.

Reply:
I don't think that will work, since the issue is not really during insert but when updating manga/chapter details from the database, the db trigger will always run when there is an update to a specific fields in this case, for manga it would be when adding and removing (favorite) value changes, during insert it defaults to 0.

It's basically used during restoring chapter/manga:

```kotlin
private suspend fun updateExistingChapters(chapters: List<Chapter>) {
        handler.await(true) {
            chapters.forEach { chapter ->
                chaptersQueries.update(
                    mangaId = null,
                    url = null,
                    name = null,
                    scanlator = null,
                    read = chapter.read,
                    bookmark = chapter.bookmark,
                    lastPageRead = chapter.lastPageRead,
                    chapterNumber = null,
                    sourceOrder = null,
                    dateFetch = null,
                    dateUpload = null,
                    chapterId = chapter.id,
                    version = chapter.version,
                    isSyncing = chapter.isSyncing,
                )
            }
        }
    }
``` 


```kotlin
        return if (isSync) {
            chapter.copy(
                id = dbChapter.id,
                bookmark = chapter.bookmark || dbChapter.bookmark,
                read = chapter.read,
                lastPageRead = chapter.lastPageRead,
                isSyncing = 1,
            )
        } else {
            chapter.copyFrom(dbChapter).let {
                when {
                    dbChapter.read && !it.read -> it.copy(read = true, lastPageRead = dbChapter.lastPageRead)
                    it.lastPageRead == 0L && dbChapter.lastPageRead != 0L -> it.copy(
                        lastPageRead = dbChapter.lastPageRead,
                    )
                    else -> it
                }
            }
        }
```


I need to set the isSyncing "flag" to 1 that way it disables the trigger for that particular manga/chapter temporarily. If there is a better way to disable a trigger in sqlite do let me know.

Reply:
What i'm proposing is something like the code below instead of making it part of the Manga or Chapter class. It's fine to keep it in db. Will this not work?
```kt
private suspend fun updateExistingChapters(chapters: List<Chapter>, isSyncing: Boolean) {
    handler.await(true) {
        chapters.forEach { chapter ->
            chaptersQueries.update(
                mangaId = null,
                url = null,
                name = null,
                scanlator = null,
                read = chapter.read,
                bookmark = chapter.bookmark,
                lastPageRead = chapter.lastPageRead,
                chapterNumber = null,
                sourceOrder = null,
                dateFetch = null,
                dateUpload = null,
                chapterId = chapter.id,
                version = chapter.version,
                isSyncing = isSyncing.toLong(),
            )
        }
    }
}
    ```

Reply:
Oh yea, that should work, apologies misunderstood you.

Reply:
> Some class still has that field

Oh which one do you mean? I know it's in the Manga and Chapter.kt

> Some class still has that field

Okay I think it's done now, let me know if that's all?